test/ecto/multi_test.exs

```diff
@@ -196,6 +196,18 @@ defmodule Ecto.MultiTest do
     assert [{:comments, {:insert_all, Comment, [[x: 2]], []}}] = multi.operations
   end
 
+  test "insert_all fun" do
+    fun_schema = fn _changes -> {:ok, Comment} end
+    fun_entries = fn _changes -> {:ok, [[x: 2]]} end
```

Comment:
This is definitely no longer a function but the tests are passing. This probably means we are missing some guards in the code.